app/auth/views.py

```diff
@@ -0,0 +1,154 @@
+                    return make_response(jsonify(response))
+                user = User(name=name, email=email, password=password)
+                user.save_user()
+                token_ = user.generate_token(user.id)
+                response = {
+                    'message': 'successfully registered',
+                    'token': token_.decode()
+                }
+                return make_response(jsonify(response)), 201
+            return make_response(
+                jsonify({'message': 'invalid name or email'})), 400
+        return make_response(jsonify({'message': 'empty inputs'})), 400
+
+
+class Login(MethodView):
+    """Login class handles registered user login."""
+
+    def post(self):
+        """Login user."""
+        json_dict = request.get_json()
+        email = json_dict.get('email')
+        password = json_dict.get('password')
+        user = User.query.filter_by(email=email).first()
+        if not user:
+            response = {
+                'message': 'you are not registered,please register'
+            }
+            return make_response(jsonify(response)), 400
+        elif user and user.validate_password(password):
+            token_ = user.generate_token(user.id)
+            if token_:
+                response = {
+                    'message': 'you are successfully login',
+                    'token': token_.decode()
+                }
+                return make_response(jsonify(response)), 200
+        else:
+            response = {
+                'message': 'wrong email or password, please try again'
+            }
+            return make_response(jsonify(response)), 400
+
+
+class Logout(MethodView):
+    """Logout class"""
+
+    def post(self):
+        """Logout a registered user."""
+        auth_token = request.headers.get('Authorization')
+        if auth_token:
+            resp = User.decoding_token(auth_token)
+            return make_response(
+                jsonify({'message': 'successfully logout'})), 200
+        return make_response(
+            jsonify({'messsage': 'wrong token,Please check you token if correct'})), 403
+
+
+class ResetPassword(MethodView):
+    """ResetPassword class handles resetting password."""
+
+    def post(self):
+        """Resets user password"""
+        json_dict = request.get_json()
+        email = json_dict.get('email')
+        new_password = json_dict.get('new_password')
+        user = User.query.filter_by(email=email).first()
+        if user:
+            user.password = generate_password_hash(new_password)
+            user.save_user()
+            response = {
+                "message": 'Password reset successful',
+                'new password': new_password
+            }
+            return make_response(jsonify(response)), 201
+        response = {'message': 'Invalid email'}
+        return make_response(jsonify(response)), 400
+
+
+class ChangePassword(MethodView):
```

Comment:
what's the use of this class if you have the rest password, if they perform the same tasks, you can combine them

Reply:
The ResetPassword class and ChangePassword class are not the same.
ChangePassword is when the user can remember the password but want to change the password
ResetPassword is when a user has completely forgotten the password